### PR TITLE
Unvendor Mumps.

### LINKED ISSUE #48: Un-vendor ASL and Mumps
Issue:

It seems that ASL and Mumps are being vendored with this package. I  don't think that is what we want to be doing. ASL and Mumps should have their own feedstocks where they are being built as shared libraries and we depend on them. These are the lines in the `meta.yaml` that download the two pieces of software:

```
  - folder: ASL
    url: https://github.com/coin-or-tools/ThirdParty-ASL/archive/releases/1.4.3.tar.gz
    sha256: 0a4cfa7c6c5df72d8e1c4c3ce74633421f807115a679e85cb5569c41df9a9aaf

  - folder: Mumps
    url: https://github.com/coin-or-tools/ThirdParty-Mumps/archive/releases/1.6.2.tar.gz
    sha256: 0cfdd30b9cb42d9539ad9c495629f56dd1fc0b59cea691bbe71a129fabd8895d
    patches:
      - thirdparty-mumps-linux-configure.patch  # [linux]
```

It looks like I approved them in #40 but I can't remember why I did. Maybe I was thinking we were only vendoring for windows?

Regardless, I think this needs to be undone and we setup the proper dependencies on the other conda-forge packages.

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

```
</details>

### PR DESCRIPTION
**Pull request by moorepants**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Partially fixes #48.

Just trying to see if this works.

### CODE REVIEW

**Comment by moorepants** [comment]
@conda-forge-admin, please rerender

**Comment by jschueller** [comment]
-1
tests fail for me when I dont build their mumps version
I agree it could at least be moved to a separate feedstock (coinor-mumps ?)

**Comment by moorepants** [comment]
`make test` fails with:
```
 ---- 8< ----  End of test program output  ---- 8< ----
 
    ******** Test FAILED! ********
Test program existed with nonzero status.
make[1]: *** [test] Error 1
make[1]: Leaving directory `$SRC_DIR/Ipopt/build/test'
make: *** [unitTest] Error 2
Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.8/site-packages/conda_build/cli/main_build.py", line 474, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.8/site-packages/conda_build/cli/main_build.py", line 463, in execute
    outputs = api.build(args.recipe, post=args.post, test_run_post=args.test_run_post,
  File "/opt/conda/lib/python3.8/site-packages/conda_build/api.py", line 186, in build
    return build_tree(
  File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 3008, in build_tree
    packages_from_this = build(metadata, stats,
  File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 2131, in build
    utils.check_call_env(cmd, env=env, rewrite_stdout_env=rewrite_env,
  File "/opt/conda/lib/python3.8/site-packages/conda_build/utils.py", line 410, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/opt/conda/lib/python3.8/site-packages/conda_build/utils.py", line 390, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-o', 'errexit', '/home/conda/feedstock_root/build_artifacts/ipopt_1604303939148/work/conda_build.sh']' returned non-zero exit status 2.

##[error]Bash exited with code '1'.
```

**Comment by moorepants** [comment]
Failed tests:
```
Running unitTests...
 
Testing AMPL Solver Executable...
 
 ---- 8< ----  End of test program output  ---- 8< ----
 
    ******** Test FAILED! ********
Test program existed with nonzero status.
Testing sIpopt Example redhess_cpp...
./run_unitTests: line 19: 17342 Segmentation fault      (core dumped) $@ > tmpfile 2>&1
0 
 ---- 8< ---- Start of test program output ---- 8< ----
Exception of type: OPTION_INVALID in file "../../../src/Algorithm/IpAlgBuilder.cpp" at line 268:
 Exception message: Selected linear solver MA27 not available.
Tried to obtain MA27 from shared library "libhsl.so", but the following error occured:
libhsl.so: cannot open shared object file: No such file or directory

EXIT: Invalid option encountered.
 ---- 8< ----  End of test program output  ---- 8< ----
 
    ******** Test FAILED! ********
Test program existed with nonzero status.
make[1]: *** [test] Error 1
make[1]: Leaving directory `$SRC_DIR/Ipopt/build/test'
make: *** [unitTest] Error 2
Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.8/site-packages/conda_build/cli/main_build.py", line 474, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.8/site-packages/conda_build/cli/main_build.py", line 463, in execute
    outputs = api.build(args.recipe, post=args.post, test_run_post=args.test_run_post,
  File "/opt/conda/lib/python3.8/site-packages/conda_build/api.py", line 186, in build
    return build_tree(
  File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 3008, in build_tree
    packages_from_this = build(metadata, stats,
  File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 2131, in build
    utils.check_call_env(cmd, env=env, rewrite_stdout_env=rewrite_env,
  File "/opt/conda/lib/python3.8/site-packages/conda_build/utils.py", line 410, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/opt/conda/lib/python3.8/site-packages/conda_build/utils.py", line 390, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-o', 'errexit', '/home/conda/feedstock_root/build_artifacts/ipopt_1604303939148/work/conda_build.sh']' returned non-zero exit status 2.

##[error]Bash exited with code '1'.
```

**Comment by moorepants** [comment]
There are segmentation faults as mentioned in #39.

**Comment by jschueller** [comment]
I meant, there's no way to use upstream mumps (yet?) this will fail no matter what, you can ask upstream.

**Comment by moorepants** [comment]
I'd like to see how Ipopt patches Mumps. If that isn't extensive, maybe we can add the patch to the mumps-feedstock with no adverse affects on other packages that depend on mumps. The option to create a coinor-mumps package on conda-forge could work too. Maybe that is the right solution.

**Comment by jschueller** [comment]
also, last time I checked ipopt doesnt even look for an external mumps, it only looks for coin-mumps, so maybe that's what happens here, an exception is thrown when no sparse solver is available

**Comment by moorepants** [comment]
> last time I checked ipopt doesnt even look for an external mumps

Good to know. Maybe I'm chasing up the wrong tree here. I can look into ipopt's current build options and see if you can link to a different shared mumps.

**Comment by moorepants** [comment]
This is in the IPOPT docs:

To compile Ipopt with a precompiled MUMPS library, you need to specify the -I flag to have the compiler find the directory containing the MUMPS header files with the --with-mumps-cflags flag of Ipopt's configure, e.g.,

--with-mumps-cflags="-I$HOME/MUMPS/include"

and you also need to provide the linker flags for MUMPS with the --with-mumps-lflags flag.

**Comment by moorepants** [comment]
I'm not sure what the mumps linker flags are supposed to be. I'd like to at least have things specified correctly and see what failures occur building with the conda forge provided mumps.

**Comment by moorepants** [comment]
There is also this note in the IPOPT docs:

>      Branch mumps5 of project ThirdParty-Mumps can be used to build a a library of MUMPS 5.2.x that is usable with Ipopt. However, initial experiments on the CUTEst testset have shown that performance with MUMPS 5.2.x is worse than with Mumps 4.10.0 on average.

**Comment by moorepants** [comment]
Here is the old configure command before #39:

```
 CFLAGS="-I$PREFIX/include -I$PREFIX/include/asl -I$PREFIX/include/mumps_seq" \
  CXXFLAGS=" -m64 -I$PREFIX/include -I$PREFIX/include/asl -I$PREFIX/include/mumps_seq" \
  --with-blas-lib="-Wl,-rpath,$PREFIX/lib -L$PREFIX/lib -llapack -lblas" \
  --with-asl-lib="-Wl,-rpath,$PREFIX/lib -L$PREFIX/lib -lasl" \
  --with-mumps-lib="-Wl,-rpath,$PREFIX/lib -L$PREFIX/lib -ldmumps_seq -lmumps_common_seq -lpord_seq -lmpiseq_seq -lesmumps -lscotch -lscotcherr -lmetis -lgfortran" \
```

**Comment by moorepants** [comment]
> So maybe we have to use mumps < 5.0.

I'm getting a conda unsatisfied error trying to find a mumps-seq < 5.0. Maybe it doesn't exist.

**Comment by moorepants** [comment]
The build worked locally for me with mumps-seq and the original linker flags.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -39,7 +33,10 @@ requirements:
     - libblas
     - liblapack
     - metis
+    - mumps-seq
   run:
+    - metis
+    - mumps-seq
```
Comment (moorepants) [inline recipe/meta.yaml:39]: Not sure if they need to be here.

Comment (jschueller) [reply recipe/meta.yaml]: nope, they have run_exports

**Comment by moorepants** [comment]
Ok, this builds, tests pass and it uses `mumps-seq` from conda forge. It is ready for review. I also have #49 which builds and passes. I can add those changes here and merge as one PR or do them sequentially.

**Comment by moorepants** [comment]
@conda-forge-admin, please rerender

**Comment by moorepants** [comment]
I've merged in master and added myself to the maintainer's list if that is ok.

**Comment by moorepants** [comment]
This is ready for review.

**Comment by moorepants** [comment]
Thanks!

**Pull request merged**